### PR TITLE
Use context directly in map sub components

### PR DESCRIPTION
**Pull request by shifucun**
All the map children components can directly access the context instead of getting them through the props.

### CODE REVIEW

File: static/js/tools/map/chart_loader.tsx [hunk 1]
```diff
@@ -69,16 +65,6 @@ import { useFetchGeoJson } from "./fetcher/geojson";
 import { useFetchMapPointCoordinate } from "./fetcher/map_point_coordinate";
 import { useFetchMapPointStat } from "./fetcher/map_point_stat";
 import { PlaceDetails } from "./place_details";
-import {
```
Comment (shifucun) [inline static/js/tools/map/chart_loader.tsx:72]: These import are not used yet, so removing for now

File: static/js/tools/map/d3_map.tsx [hunk 2]
```diff
@@ -75,39 +68,133 @@ const DEBOUNCE_INTERVAL_MS = 30;
...
 export function D3Map(props: D3MapProps): JSX.Element {
+  const { placeInfo, statVar, display } = useContext(Context);
+
   const [errorMessage, setErrorMessage] = useState("");
   const [zoomTransformation, setZoomTransformation] = useState(
     DEFAULT_ZOOM_TRANSFORMATION
   );
   const chartContainerRef = useRef<HTMLDivElement>();
 
-  function replot() {
```
Comment (shifucun) [inline static/js/tools/map/d3_map.tsx:84]: Here making draw() into a useCallback() effect, so it updates based on dependency changes, and ease the need for replot().

File: static/js/tools/map/leaflet_map.tsx [hunk 3]
```diff
@@ -64,46 +69,25 @@ export function LeafletMap(props: LeafletMapProps): JSX.Element {
     });
   }, []);
 
-  // Replot when data changes.
-  // TODO: handle resizing of chart area
-  useEffect(() => {
```
Comment (shifucun) [inline static/js/tools/map/leaflet_map.tsx:69]: This is moved to below.

**Review by chejennifer — APPROVED** [review]
nice use of useCallback()!

**Pull request merged**